### PR TITLE
feat: Remove devworkspace installation from chectl after delegate devworkspace install to che-operator

### PR DESCRIPTION
**Pull request by flacatus**
Signed-off-by: Flavius Lacatusu <flacatus@redhat.com>

### What does this PR do?
* Delegate devworkspace controller installation to che operator by removing all dwo installation steps from chectl
* Delete DevWorkspace Che objects after che uninstallation.

Reff. issue: https://github.com/eclipse-che/che-operator/pull/731
### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [ ] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [ ] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: src/commands/server/deploy.ts [hunk 1]
```diff
@@ -251,6 +246,20 @@ export default class Deploy extends Command {
...
 
+  private isDevWorkspaceEnabled(ctx: any): boolean {
+    const crPatch = ctx.crPatch
+    if (crPatch && crPatch.spec && crPatch.spec.devWorkspace && crPatch.spec.devWorkspace.enable) {
+      return true
+    }
+
+    const customCR = ctx.customCR
+    if (customCR && customCR.spec && customCR.spec.devWorkspace && customCR.spec.devWorkspace.enable) {
+      return false
```
Comment (tolusha) [inline src/commands/server/deploy.ts:257]: Should be `true` ?

Comment (flacatus) [reply src/commands/server/deploy.ts]: my mistake

File: src/tasks/installers/common-tasks.ts [hunk 2]
```diff
@@ -127,6 +127,10 @@ export function createEclipseCheCluster(flags: any, kube: KubeHelper): Listr.Lis
       ctx.isDevfileRegistryDeployed = !(flags['devfile-registry-url'] as boolean)
 
       const cheClusterCR = ctx.customCR || ctx.defaultCR
+      if (flags['workspace-engine'] === 'dev-workspace') {
+        cheClusterCR.spec.devWorkspace.enable = true
+      }
+
```
Comment (tolusha) [inline src/tasks/installers/common-tasks.ts:133]: Should be removed?

File: src/api/kube.ts [hunk 3]
```diff
@@ -1579,6 +1579,10 @@ export class KubeHelper {
       merge(cheClusterCR, ctx.crPatch)
     }
 
+    if (flags['workspace-engine'] === 'dev-workspace') {
+      cheClusterCR.spec.devWorkspace.enable = true
+    }
+
```
Comment (tolusha) [inline src/api/kube.ts:1605]: Should be under `if (useDefaultCR) {`

**Review by tolusha — APPROVED** [review]

**Comment by nickboldt** [comment]
Does this need to be cherrypicked into 7.28.x branch for use in CRW 2.8?

**Comment by tolusha** [comment]
@nickboldt 
It isn't needed.

**Comment by nickboldt** [comment]
And yet, it's being backported anyway: https://github.com/che-incubator/chectl/pull/1176#pullrequestreview-634460187 :D

**Pull request merged**